plugins/catalog-backend/src/service/router.ts

```diff
@@ -142,7 +142,62 @@ export async function createRouter(
           );
         }
         res.status(200).json(entities[0]);
-      });
+      })
+
+      .get('/entities/by-uid/:uid/attachments/:key', async (req, res) => {
+        const { uid, key } = req.params;
+        const attachment = await entitiesCatalog.attachment(uid, key);
+
+        if (!attachment) {
+          throw new NotFoundError(
+            `No attachment with key '${key}' found for entity with uid ${uid}'`,
+          );
+        }
+
+        // TODO: Include etag for caching
+
+        res
+          .status(200)
+          .contentType(attachment.contentType)
+          .send(attachment.data);
+      })
+
+      .get(
+        '/entities/by-name/:kind/:namespace/:name/attachments/:key',
+        async (req, res) => {
+          const { kind, namespace, name, key } = req.params;
+          const { entities } = await entitiesCatalog.entities({
+            filter: basicEntityFilter({
+              kind: kind,
+              'metadata.namespace': namespace,
+              'metadata.name': name,
+            }),
+          });
+          if (!entities.length) {
+            throw new NotFoundError(
+              `No entity named '${name}' found, with kind '${kind}' in namespace '${namespace}'`,
+            );
+          }
+
+          const attachment = await entitiesCatalog.attachment(
+            entities[0].metadata.uid!,
+            key,
+          );
+
+          if (!attachment) {
+            throw new NotFoundError(
+              `No attachment with key '${key}' found for entity named '${name}', with kind '${kind}' in namespace '${namespace}'`,
+            );
+          }
+
+          // TODO: Include etag for caching
+
+          res
+            .status(200)
+            .contentType(attachment.contentType)
```

Comment:
Could have some fun with creating `application/javascript` attachments here or the like 😅. I guess as long as we keep the content type hardcoded in the processor we're good, but gotta be a bit careful and document that piece :p. Would hate to have to set up some kind of domain isolation just for this ^_>